Assets/HotmaxEditorVR/loadPropsMenu.cs

```diff
@@ -0,0 +1,288 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+public class loadPropsMenu : MonoBehaviour
+{
+    //--local refs
+    private stateManager _stateManagerMutatorRef;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
+
+    //--ref to props scene object
+    private GameObject propParent;
+
+    //--menu prop tile parent and child
+    public GameObject propTiles;
+
+    //--Lists to handle getting unique props in menu
+    private List<string> propName = new List<string>();
+    private List<GameObject> propObject = new List<GameObject>();
+    private List<List<GameObject>> chunkedList = new List<List<GameObject>>();
+
+    //--if false, disabled initial props on stage
+    public bool propIsInStageAlready = false;
+
+    private int currentPage;
+
+    //TODO: for some reason the steamVR trigger event listener won't attach in enable or Awake. works in Start...
+    private void OnEnable()
+    {
+        Invoke("waitToLoadTriggerEvent", .001f);
+        stateManager.editorModeEvent += updateEditorMode;
+    }
+
+    void waitToLoadTriggerEvent()
+    {
+        inputManager.trackedController2.TriggerClicked += triggerClicked;
+    }
+
+
+    private void OnDisable()
+    {
+        inputManager.trackedController2.TriggerClicked -= triggerClicked;
+
+        stateManager.editorModeEvent -= updateEditorMode;
+    }
+
+    void updateEditorMode(stateManager.editorModes value)
+    {
+        _editorMode = value;
+
+        if (_editorMode == stateManager.editorModes.spawnMenuMode)
+        {
+            toggleMenuComponents(true);
+
+            //--make menu face the HMD
+            GameObject menu = transform.parent.transform.gameObject;
+            GameObject vrCam = init.vrCamera;
+
+            Vector3 forwardAmount = vrCam.transform.position + (vrCam.transform.forward * 2.15f);
+            Quaternion rotationAmount = vrCam.transform.rotation;
+
+            menu.transform.position = new Vector3(forwardAmount.x, vrCam.transform.position.y - 1, forwardAmount.z);
+
+            menu.transform.rotation = rotationAmount;
+            menu.transform.localEulerAngles = new Vector3(0, menu.transform.localEulerAngles.y + 180, 0);
+        }
+        else
+        {
+            toggleMenuComponents(false);
+        }
+    }
+
+    void Start()
+    {
+        _stateManagerMutatorRef = GameObject.FindObjectOfType(typeof(stateManager)) as stateManager;
+
+        propParent = GameObject.Find("props");
```

Comment:
I would make a TODO note here that "props" is going to come from PropsClass in the form of a list, on a completion of socket event.